File: app/src/main/java/com/example/pc_3/retrofitexample/Login/LoginContract.java
package com.example.pc_3.retrofitexample.Login;

/**
 * Created by PC-3 on 09/10/2017.
 */

interface LoginContract {
    interface View {
        void showMessage(String message);

        void showProgress();

        void dismissProgress();

        void userLogged();
    }

    interface Presenter {
        void requestLogin(String user, String password);

        void userLogged();

        void showMessage(String message);
    }

    interface Model {
        void setPresenter(LoginContract.Presenter presenter);

        void requestLogin(String user, String password);
    }
}

File: app/src/main/java/com/example/pc_3/retrofitexample/Login/LoginContract.kt
package com.example.pc_3.retrofitexample.Login

/**
 * Created by PC-3 on 09/10/2017.
 */

internal interface LoginContract {
    interface View {
        fun showMessage(message: String)

        fun showProgress()

        fun dismissProgress()

        fun userLogged()
    }

    interface Presenter {
        fun requestLogin(user: String, password: String)

        fun userLogged()

        fun showMessage(message: String)
    }

    interface Model {
        fun setPresenter(presenter: LoginContract.Presenter)

        fun requestLogin(user: String, password: String)
    }
}
